src/realm/index_string.cpp

```diff
@@ -338,6 +366,15 @@ void IndexArray::from_list_all_ins(StringData upper_value, std::vector<ObjKey>&
 void IndexArray::from_list_all(const Mixed& value, std::vector<ObjKey>& result, const IntegerColumn& rows,
                                const ClusterColumn& column) const
 {
+    if (column.full_word()) {
```

Comment:
I think the `from_list_*()` methods don't work for strings in a list that are longer than s_max_offset + 4. Those strings have been truncated so the assumption that we can reconstruct the entire string from the index data doesn't hold anymore. For fulltext structures we have assumed that words are space separated, so the strings won't get that long, but for lists of strings I think we still have to handle these long ones. Could you at least add a test for this to check the behaviour?

Reply:
When we store the full word, the strings are not truncated. So in this case we will still have a problem with potential stack overflow. I don't think this is solvable in the current design. A added a test.

Reply:
I'm not confident about shipping this with a known crash waiting for someone to report it. If this feature is only intended for use cases with "short" strings in a list, what do you think about adding an exception when trying to insert "long" strings above some threshold to the index. If this limitation only applies to the new "full word" indexes then it isn't really breaking, and it is more graceful than crashing.

Reply:
Ok. Now we will throw if string is too long.